cardano-db/src/Cardano/Db/Query.hs

```diff
@@ -220,6 +221,21 @@ queryBlockNo blkNo = do
     pure (blk ^. BlockId)
   pure $ fmap unValue (listToMaybe res)
 
+queryBlockNoAndEpoch :: MonadIO m => Word64 -> ReaderT SqlBackend m (Maybe (BlockId, Word64))
+queryBlockNoAndEpoch blkNo = do
+  res <- select $ do
+    blk <- from $ table @Block
+    where_ (blk ^. BlockBlockNo ==. just (val blkNo))
+    pure (blk ^. BlockId, blk ^. BlockEpochNo)
+  pure $ convert (listToMaybe res)
+  where
+  convert :: Maybe (Value (Key Block), Value (Maybe Word64)) -> Maybe (BlockId, Word64)
+  convert mr =
+    case mr of
+      Nothing -> Nothing
+      Just (_, Value Nothing) -> Nothing -- Should not ever happen.
```

Comment:
Personal note this can't happen because it's only null for genesis which doesn't have a blockNo in the first place.